py-nbclassic: add v1.1.0

dmagdavector opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

dmagdavector committed:
py-nbclassic: add v1.1

tldahlgren commented on var/spack/repos/builtin/packages/py-nbclassic/package.py:
```diff
@@ -14,11 +14,13 @@ class PyNbclassic(PythonPackage):
 
+    depends_on("python@3.7:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
     depends_on("py-jupyter-packaging@0.9:0", when="@0.3.3:", type="build")
```
If I'm reading this correctly, `py-jupyter-packaging` can be version 1, from `pyproject.toml`:

```
[build-system]
requires = [
  "jupyter_packaging~=0.9,<2",
  "jupyter_server>=1.17.0",
  "babel"
]
```

tldahlgren replied:
There appear to be fewer other dependencies per `setup.cfg`:

```
python_requires = >=3.7
install_requires =
    ipython_genutils
    notebook_shim>=0.2.3
    nest-asyncio>=1.5
    ipykernel # bless IPython kernel for now
```

dmagdavector replied:
`py-jupyter-packaging` version 1 does not exist:
* https://github.com/jupyter/jupyter-packaging

The build dependencies are unchanged since 0.4.8, the version currently available in Spack:

```
[build-system]
requires = ["jupyter_packaging~=0.9,<2", "babel"]
build-backend = "jupyter_packaging.build_api"
```
* https://github.com/jupyter/nbclassic/blob/v0.4.8/pyproject.toml

So since they already work, felt no change was necessary.

As for the stuff in `setup.cfg`, there are already `depends_on()` lines for those, it turns out that many that are listed are 'excessive' in that they are not for `py-nbclassic` itself, but for a sub-dependency. I've tweaked things with a new/updated commit that notes that change for v1.1.

tldahlgren reviewed (changes requested):
Confirmed the version sha256. It looks like the dependencies needs to be tweaked if they are to match `pyproject.toml` and `setup.cfg`

dmagdavector committed:
py-nbclassic: reduce explicit dependencies for v1.1.0

Having all the 'excess' packages listed did not break anything, as
they were needed for `py-jupyter-server` (pulled in via `py-notebook-shim`)
anyway, but the change makes it more clear on why things are being pulled in.
